src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int LIMIT = 10;
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        Figure[] figures = new Figure[]
+                {new Circle(colorSupplier.getRandomColor(), random.nextInt(LIMIT)),
+                new RightTriangle(colorSupplier.getRandomColor(),
+                        random.nextInt(LIMIT), random.nextInt(LIMIT)),
+                new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(LIMIT), random.nextInt(LIMIT),random.nextInt(LIMIT)),
+                new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(LIMIT), random.nextInt(LIMIT)),
+                new Square(colorSupplier.getRandomColor(), random.nextInt(LIMIT))};
+
```

Comment:
let's use switch/case here, create a random number (from 0 to 3 and use default case for the last figure type) and depending on that number you'll use switch case to create a random figure.